File: src/go/postprompt/controlState.go
package postprompt

type Phase int
type SuperPhase int

const (
	DrawPhase Phase = iota
	PrePhase
	MainPhase
	EndPhase
	StepOnePhase = 1
)

const (
	PreSuperPhase SuperPhase = iota
	MainSuperPhase
	DoneSuperPhase
	StepOneSuperPhase = 1
)

type ControlState struct {
	uids [2]int
	phase Phase
	superPhase SuperPhase
	turnOwner int
	hasDrawn bool
}

func NewControlState(uid1, uid2 int) *ControlState {
	controlState := new(ControlState)
	controlState.uids = [2]int{uid1,uid2}
	controlState.phase = 0
	controlState.superPhase = 0
	controlState.turnOwner = uid1
	controlState.hasDrawn = false
	return controlState
}

File: src/client/model/control_state.py
from pplib import json_parser

class Control_State:
	def __init__(self,obj):
		self.super_phase = json_parser.get_int(obj,'superPhase')
		self.phase = json_parser.get_int(obj,'phase')
		self.turn_owner = json_parser.get_int(obj,'turnOwner')
		self.has_drawn = json_parser.get_bool(obj,'hasDrawn')
